src/EditorFeatures/Test/MetadataAsSource/MetadataAsSourceTests.cs

```diff
@@ -1627,5 +1627,338 @@ Class C
+        {
+            var metadataSource = @"
+public ref struct S
+{
+}
+";
+            var symbolName = "S";
+
+            await GenerateAndVerifySourceAsync(metadataSource, symbolName, LanguageNames.CSharp, $@"#region {FeaturesResources.Assembly} ReferencedAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null
+// {CodeAnalysisResources.InMemoryAssembly}
+#endregion
+
+public ref struct [|S|]
+{{
+}}");
+
+            await GenerateAndVerifySourceAsync(metadataSource, symbolName, LanguageNames.VisualBasic, $@"#Region ""{FeaturesResources.Assembly} ReferencedAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null""
+' {CodeAnalysisResources.InMemoryAssembly}
+#End Region
+
+Imports System
+Imports System.Runtime.CompilerServices
+
+<IsByRefLikeAttribute> <Obsolete(""Types with embedded references are not supported in this version of your compiler."", True)>
+Public Structure [|S|]
+End Structure");
+        }
+
+        [WorkItem(34650, "https://github.com/dotnet/roslyn/issues/34650")]
+        [Fact, Trait(Traits.Feature, Traits.Features.MetadataAsSource)]
+        public async Task TestReadOnlyRefStruct()
+        {
+            var metadataSource = @"
+public readonly ref struct S
+{
+}
+";
+            var symbolName = "S";
+
+            await GenerateAndVerifySourceAsync(metadataSource, symbolName, LanguageNames.CSharp, $@"#region {FeaturesResources.Assembly} ReferencedAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null
+// {CodeAnalysisResources.InMemoryAssembly}
+#endregion
+
+public readonly ref struct [|S|]
+{{
+}}");
+
+            await GenerateAndVerifySourceAsync(metadataSource, symbolName, LanguageNames.VisualBasic, $@"#Region ""{FeaturesResources.Assembly} ReferencedAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null""
+' {CodeAnalysisResources.InMemoryAssembly}
+#End Region
+
+Imports System
+Imports System.Runtime.CompilerServices
+
+<IsByRefLikeAttribute> <IsReadOnlyAttribute> <Obsolete(""Types with embedded references are not supported in this version of your compiler."", True)>
+Public Structure [|S|]
+End Structure");
+        }
+
+
+        [WorkItem(34650, "https://github.com/dotnet/roslyn/issues/34650")]
+        [Fact, Trait(Traits.Feature, Traits.Features.MetadataAsSource)]
+        public async Task TestReadOnlyMethod()
+        {
+            var metadataSource = @"
+public struct S
+{
+    public readonly void M() {}
+}
+";
+            var symbolName = "S.M";
+
+            await GenerateAndVerifySourceAsync(metadataSource, symbolName, LanguageNames.CSharp, languageVersion: "Preview",
+                expected: $@"#region {FeaturesResources.Assembly} ReferencedAssembly, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null
+// {CodeAnalysisResources.InMemoryAssembly}
+#endregion
+
+public struct S
+{{
+    public readonly void [|M|]();
+}}");
+
+            // TODO: how to use preview features in metadata source but display in VB?
```

Comment:
Would appreciate guidance from anyone who knows how to do this :)

Reply:
Sorry, what's the question? It doesn't work like your other VB tests?

Reply:
When I have a call like this:
```cs
await GenerateAndVerifySourceAsync(metadataSource, symbolName, LanguageNames.CSharp, languageVersion: "Preview", ...);
```
It's able to use the CSharp LanguageVersion.Preview for both the metadata source and the expected source. But when I change to:

```cs
await GenerateAndVerifySourceAsync(metadataSource, symbolName, LanguageNames.VisualBasic, languageVersion: "Preview", ...);
```

I'm not able to specify a C#-specific language version any more--it'll just get confused because no such VB language version exists. It always uses the default C# version for the metadata source, so I get an error for using a preview C# language feature.

Reply:
I'm just gonna modify the test harness to accept a `metadataLanguageVersion`.